cmd/lava/internal/run/run.go

```diff
@@ -447,8 +447,8 @@ func writeOutputs(rep engine.Report) (report.ExitCode, error) {
 		return 0, fmt.Errorf("render report: %w", err)
 	}
 
-	if reportConfig.Metrics != "" {
-		if err = metrics.WriteFile(reportConfig.Metrics); err != nil {
+	if config.Get(reportConfig.Metrics) != "" {
+		if err = metrics.WriteFile(config.Get(reportConfig.Metrics)); err != nil {
```

Comment:
You can avoid one call to `config.Get` with:

```suggestion
	if metricsFile := config.Get(reportConfig.Metrics); metricsFile != "" {
		if err = metrics.WriteFile(metricsFile); err != nil {
```